tools/xdp/s2n-quic-xdp/src/io/tx.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    if_xdp::{RxTxDescriptor, UmemDescriptor},
+    umem::Umem,
+};
+use core::task::{Context, Poll};
+use s2n_codec::{Encoder as _, EncoderBuffer};
+use s2n_quic_core::{
+    event,
+    io::tx,
+    sync::spsc,
+    xdp::{encoder, path},
+};
+
+pub type Free = spsc::Receiver<UmemDescriptor>;
+pub type Occupied = spsc::Sender<RxTxDescriptor>;
+
+pub struct Tx {
+    free: Free,
+    occupied: Occupied,
+    umem: Umem,
+    encoder: encoder::State,
+    is_full: bool,
+}
+
+impl Tx {
+    /// Creates an TX IO interface for a s2n-quic endpoint
```

Comment:
```suggestion
    /// Creates a TX IO interface for an s2n-quic endpoint
```

Reply:
I accidentally did an `--amend` instead of another commit so sorry about the diff size.